src/sentry/options/types.py

```diff
@@ -0,0 +1,173 @@
+"""
+sentry.options.types
+~~~~~~~~~~~~~~~~~~~~
+
+:copyright: (c) 2010-2015 by the Sentry Team, see AUTHORS for more details.
+:license: BSD, see LICENSE for more details.
+"""
+from __future__ import absolute_import, print_function
+
+from sentry.utils.yaml import safe_load
+from yaml.parser import ParserError
+from yaml.scanner import ScannerError
+
+__all__ = (
+    'InvalidTypeError',
+    'Any', 'Bool', 'Int', 'Float', 'String', 'Dict', 'Sequence',
+)
+
+
+class InvalidTypeError(TypeError):
+    pass
+
+
+class OptionType(object):
+    """Base OptionType that provides type coersion"""
+    name = ''
+    # Default value to be returned when initializing
+    default = None
+    # Types that do not need to be coerced
+    expected_types = (object,)
+    # Types that are acceptable for coersion
+    compatible_types = (basestring,)
+
+    def __call__(self, value=None):
+        if value is None:
+            return self._default()
+        if self.test(value):
+            return value
+        if isinstance(value, self.compatible_types):
+            rv = self.convert(value)
+            # Make sure convert was able to do the right thing
+            # and give us the type we were expecting
+            if self.test(rv):
+                return rv
```

Comment:
I was trying to encapsulate the redundant "check if the type is correct after conversion" all int he same place instead of repeating in every type.

This problem surfaces more heavily with the ones that accept json/yaml since the json/yaml gets parsed, then it's maybe the wrong type after.

I chose to do it here for everything and it made the convert functions simpler as a tradeoff. Then it also avoided repeating the same 'raise InvalidTypeError' logic everywhere along with the repeated formatting, etc.